Add GitHub authentication instructions, resolves #505 (#508)
railsgirls/railsgirls.github.io: _pages/github.md

Click "Generate token" at the bottom of the page.

On the next page you'll see your PAT. This is the only time you'll have access to it, so don't click away from this page until you have completed the _push_ step in the next section. 

Copy and save the PAT token, ideally in a secure password manager. Be careful not to copy and spaces before or after the token -- you can use the two-squares copy button at the end of the token to be certain. You can keep the browser window open until you've completed the next step.

## Push your app to GitHub using the command line (part 2)

Now we want to _push_ to local changes in the Git repository to repository on GitHub with the following command in your terminal.

{% highlight sh %}
git push -u origin master
{% endhighlight %}

When the authentication prompt appears in your terminal, use your PAT as the password, example below. Note that when you paste your PAT in the password, it will not show. Don't paste again, or you will be entering the token twice.

{% highlight sh %}
Username: <your GitHub username>
Password: <paste in your personal access token>
{% endhighlight %}

> You will need your PAT every time you want to push your code, or you can save the PAT in your computer. This process varies per operating system, so your coach can help you with this process if you plan to keep pushing your code to GitHub. 

Reviewer: I always use the SSH method to push code so I'm unfamiliar with all the details, but if I configure a repo remote using HTTPS, I don't appear to have to authenticate on every push, only the one time 🤔 
There's probably some caching going on on macOS already (using Homebrew Git). It might differ per OS even.

We can update the text to be a bit less final on the expected result:

```suggestion
You may need your PAT every time you want to push your code, or you can save the PAT on your computer. This process varies per operating system, so your coach can help you with this process if you plan to keep pushing your code to GitHub.
```
